### ISSUE TITLE
Invalid height attributes result in a component with a height of zero pixels

### ISSUE THREAD

**Issue by GuyGallant1** [stephanrauh/ngx-extended-pdf-viewer#823]
Hi,
I have a problem with ngx-extended-pdf-viewer (9.0.5). Currently I can compile my angular 12 code (with ngx-extended-pdf-viewer) and run the project. However when I try to display the PDF it has not been rendered and there are not errors in the console.

Here are a couple of snippets of code:
**Package.json**:  "ngx-extended-pdf-viewer": "9.0.5",
**HTML**
```html
           <ngx-extended-pdf-viewer
                [base64Src]="documentUrl"
                [useBrowserLocale]="useBrowserLocale" 
                [textLayer]="textLayer" 
                [logLevel]="0" 
                height="'60vh'"> 
            </ngx-extended-pdf-viewer> 
```  
TypeScript code:
```typescript
import { pdfBase64 }                                         from '@shared/pdfBase64';       //same as your demo project

  ngOnInit() { 
        let vm =  this;
         vm.documentUrl = pdfBase64;         // contents
      
    }
```
Is there a debug parameter? At the moment I have no method to determine on why the rendering is not working.
Thanks

**Comment by stephanrauh** [comment]
Strange. You can add the parameter `[minifiedJSLibraries]="false"`. That make debugging a lot easier.

If you don't find out what's going on, you can send me a reproducer. Maybe I can find out what's wrong.

**Comment by GuyGallant1** [comment]
Here is a screen shot of the installed pdf javascript libraries.
![Capture](https://user-images.githubusercontent.com/32422220/124496521-8664ab80-dd87-11eb-9b38-59306bac9d0d.PNG)

Meanwhile I will be testing [minifiedJSLibraries]="false" in a few minutes

**Comment by GuyGallant1** [comment]
Here is another screen shot . This screen shot shows the pdf (from your demo)  after it populates [base64Src]="documentUrl"
![Capture](https://user-images.githubusercontent.com/32422220/124497054-4c47d980-dd88-11eb-8eb5-c92dd6f6e517.PNG)

**Comment by GuyGallant1** [comment]
I have rebuild a simple project. In this project I can reproduce the problem. Hopefully you can find out if its an environment problem or whatever. I seem unable to add my compress file (format 7-zip). Is it possible to email you the test project

**Comment by stephanrauh** [comment]
Yeah, sure. Drop me an email at webmaster1@beyondjava.de.

BTW: currently I'm on sick leave, and I don't feel well indeed. Please apologize if I need more time than usual to examine your project and to find the bug.

Best regards,
Stephan

**Comment by stephanrauh** [comment]
Thanks for sending me a reproducer! After starting it, it was just a matter of a few minute to spot the bug. Great work!

As it turns out, you've passed an invalid height parameter. It should either be `[height]="'60vh'"` or `height="60vh"` (note the cute little ticks). For some reason, ngx-extended-pdf-viewer translates that as a height of zero pixels. So it's not surprising you can't see the PDF file.

I've decided to add an additional check. A height of zero pixels should raise an exception.

**Comment by stephanrauh** [comment]
BTW, I've improved the algorithm setting a sensible height automatedly. Try omitting the `[height]` attribute. Chances are you're happy with the result. The idea is to use as much screen estate as possible without showing a scrollbar.

**Comment by GuyGallant1** [comment]
Good morning (here at least),

Will you be releasing a new update to 9.0.5 (ie 9.0.6)?  Also is your 
software written in pure Javascript?

Guy

**Comment by stephanrauh** [comment]
Good evening, :)

I plan to dedicate some time to ngx-extended-pdf-viewer on Saturday. That should result in a new release. I'm not sure about the version number yet. This week, developers had a couple of interesting ideas which justify a new major version. So it's possible I'm going to publish a few alpha or beta versions.

ngx-extended-pdf-viewer is an Angular component, so it's written in TypeScript. But it's based on pdf.js, which is a pure JavaScript project. Recently, they've upgraded to a very new dialect of JavaScript, which fails to run on not-so-old browsers such as Firefox 68 (the previous ESR release of Firefox).

Cheers,
Stephan

**Comment by stephanrauh** [comment]
The additional check has landed with version 10.0.0-alpha.1.
